agent/agent.go

```diff
@@ -2451,6 +2471,34 @@ func (a *Agent) validateService(service *structs.NodeService, chkTypes []*struct
 		}
 	}
 
+	// Check IPv4/IPv6 tagged addresses
+	if service.TaggedAddresses != nil {
```

Comment:
The nil check here should not be necessary as `val, ok := service.TaggedAddresses[...]` will work fine even on a nil map.

This shouldn't need to block merging though.